### PR TITLE
Updated repository config files

### PR DESCRIPTION
**Pull request by micahhausler**
Signed-off-by: Micah Hausler <mhausler@amazon.com>

## Description

Extracted repo config changes from #563

## Why is this needed

* Go version is out of date in `ci.yaml`
* `.mk` files were missing from editor config
* Coverage was counted against for generated files

Fixes: #

N/A

## How Has This Been Tested?

Ran verify/tests locally

## How are existing users impacted? What migration steps/scripts do we need?

N/A

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] added unit or e2e tests
- [x] provided instructions on how to upgrade

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -25,11 +25,13 @@ jobs:
...
         with:
-          go-version: "1.14.6"
+          go-version: "1.17"
       - name: Generate
         run: make generate
+      - name: go clean -testcache
+        run: go clean -testcache
       - name: go test
-        run: go test -coverprofile=coverage.txt ./...
+        run: go test -coverprofile=coverage.txt ./... -v
```
Comment (mmlb) [inline .github/workflows/ci.yaml:34]: can you drop this one? I'm not sure we really need it. I don't really care to see the test names when there's no failure (I don't even look at the test run if there's no failure tbh). Go does the right thing when there's failure and only expands the failing tests. And actually having `-v` in there just adds a ton of noise when there is a test, printing out the names of all the passing tests.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -25,11 +25,13 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v2
         with:
-          go-version: "1.14.6"
+          go-version: "1.17"
       - name: Generate
         run: make generate
+      - name: go clean -testcache
+        run: go clean -testcache
```
Comment (mmlb) [inline .github/workflows/ci.yaml:32]: Do we really need this? If so why?

Comment (micahhausler) [reply .github/workflows/ci.yaml:32]: sure, It was in the Makefile, so I added for consistency

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (micahhausler) [reply .github/workflows/ci.yaml:34]: sure, It was in the Makefile, so I added for consistency

Comment (mmlb) [reply .github/workflows/ci.yaml:34]: ok makes sense. consistency > sense :D

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (mmlb) [reply .github/workflows/ci.yaml:32]: ok makes sense. consistency > sense :D

**Review by mmlb — APPROVED** [review]

**Pull request merged**